src/tasks.py

```diff
@@ -140,6 +194,36 @@ def execute_yara(self, job: Job, files: List[str]) -> None:
                 f"in {scanned_datasets}/{job.total_datasets} ({dataset_percent:.0%}) of datasets.",
             )
 
+    def get_match_context(
+        self, data: bytes, matches: List[yara.Match]
+    ) -> dict:
+        context = {}
+        for yara_match in matches:
+            match_context = []
+            for string_match in yara_match.strings:
+                expression_keys = []
```

Comment:
Since this is only used for presence checking, this should be a set instead of a list. But I think it's not necessary at all (see next comment)